From e7244a3b70a6a76bb2943e8a5061b0fff28bb873 Mon Sep 17 00:00:00 2001
From: Honore Tricot <honorex.tricot@intel.com>
Date: Tue, 21 Aug 2012 17:49:43 +0200
Subject: [PATCH 17/78] Revert "[PORT FROM R3] WLAN: fix provisioning issue
 when change MAC address without calibration"

BZ: 53052

port from R3 is buggy

This reverts commit d86c496ab0a77c0b6f0dda187575f218737e23d9.

Change-Id: I359a022b8c7c035669417ea934f8f9774df86084
Signed-off-by: Honore Tricot <honorex.tricot@intel.com>
Reviewed-on: http://android.intel.com:8080/63035
Reviewed-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77974
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wlan_prov/wlan_provisioning.c |   24 ------------------------
 1 files changed, 0 insertions(+), 24 deletions(-)

diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index 3e9cab3..71422d3 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -237,7 +237,6 @@ int main(int argc, char **argv)
 	unsigned char *ChaabiMacAddr = NULL;
 	int res = 0;
 	char device_id[64];
-	int unbind_bind_request = 0;
 
 	/* Check parameters */
 	if (argc != 1) {
@@ -263,7 +262,6 @@ int main(int argc, char **argv)
 
 	if (!nvsBinFile) {
 		LOGI("run calibration");
-		unbind_bind_request = 1;
 		if (wifi_calibration()) {
 			res =  -2;
 			goto end;
@@ -309,8 +307,6 @@ int main(int argc, char **argv)
 			res =  -4;
 			goto end;
 		}
-		LOGI("MAC updated");
-		unbind_bind_request = 1;
 	}
 
 	/*
@@ -333,26 +329,6 @@ int main(int argc, char **argv)
 
 end:
 
-	if (unbind_bind_request) {
-		LOGI("unbind/bind the driver");
-		/*
-		* If we are not allowed to bind/unbind the driver, the mac address as
-		* well as the new configuration firmware (.nvs) will not be taken into
-		* account. In that case, the reboot is required after flashing for the
-		* time the board.
-		*/
-		if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)
-			|| unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
-			|| bind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)) {
-			/*
-			* Rebooting the board: In this level, the NVS was saved. The
-			* next reboot will be fine since no need to bind/unbind the
-			* driver.
-			*/
-			goto fatal;
-		}
-	}
-
 	/* Take into acount the new NVS firmware */
 	sync();
 
-- 
1.7.9

